From 3cac3e6555223e2103eab94a79b5031d4c2ab6a6 Mon Sep 17 00:00:00 2001
From: Rotariu Marian Cristian-B44860 <marian.rotariu@freescale.com>
Date: Wed, 8 Oct 2014 15:48:36 +0300
Subject: [PATCH 881/987] dpaa_eth: remove unnecessary conversion

Protocol and nexthdr, variables from ipv4 and ipv6 layers, are 1 byte long,
therefore no byte order conversion is needed.

Change-Id: I487b044d9ad942cf39ccbb500318ae8d7d0d9e08
Signed-off-by: Rotariu Marian Cristian-B44860 <marian.rotariu@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/20934
Reviewed-by: Madalin-Cristian Bucur <madalin.bucur@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/32493
[Yang: Original patch taken from
QorIQ-SDK-V1.8-SOURCE-20150619-yocto.iso]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
index 81a240d..48b8405 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
@@ -1618,8 +1618,8 @@ int dpa_enable_tx_csum(struct dpa_priv_s *priv,
 	fm_prs_result_t *parse_result;
 	struct iphdr *iph;
 	struct ipv6hdr *ipv6h = NULL;
-	int l4_proto;
-	int ethertype = ntohs(skb->protocol);
+	u8 l4_proto;
+	u16 ethertype = ntohs(skb->protocol);
 	int retval = 0;
 
 	if (skb->ip_summed != CHECKSUM_PARTIAL)
@@ -1651,13 +1651,13 @@ int dpa_enable_tx_csum(struct dpa_priv_s *priv,
 		parse_result->l3r = FM_L3_PARSE_RESULT_IPV4;
 		iph = ip_hdr(skb);
 		DPA_BUG_ON(iph == NULL);
-		l4_proto = ntohs(iph->protocol);
+		l4_proto = iph->protocol;
 		break;
 	case ETH_P_IPV6:
 		parse_result->l3r = FM_L3_PARSE_RESULT_IPV6;
 		ipv6h = ipv6_hdr(skb);
 		DPA_BUG_ON(ipv6h == NULL);
-		l4_proto = ntohs(ipv6h->nexthdr);
+		l4_proto = ipv6h->nexthdr;
 		break;
 	default:
 		/* We shouldn't even be here */
-- 
1.9.1

